ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListSourceExecutor.java

```diff
@@ -191,14 +200,42 @@ private static SourceDescriptionWithWarnings describeSource(
       ), statementText);
     }
 
+    List<RunningQuery> sourceQueries = getQueries(ksqlEngine,
+        q -> q.getSourceNames().contains(dataSource.getName()));
+    List<RunningQuery> sinkQueries = getQueries(ksqlEngine,
+        q -> q.getSinkName().equals(dataSource.getName()));
+
     Optional<org.apache.kafka.clients.admin.TopicDescription> topicDescription = Optional.empty();
+    Optional<ConsumerGroupDescription> consumerGroupDescription = Optional.empty();
+    Map<TopicPartition, OffsetAndMetadata> topicAndConsumerOffsets = new LinkedHashMap<>();
+    Map<TopicPartition, ListOffsetsResultInfo> topicAndStartOffsets = new LinkedHashMap<>();
+    Map<TopicPartition, ListOffsetsResultInfo> topicAndEndOffsets = new LinkedHashMap<>();
     final List<KsqlWarning> warnings = new LinkedList<>();
     if (extended) {
       try {
         topicDescription = Optional.of(
             serviceContext.getTopicClient().describeTopic(dataSource.getKafkaTopicName())
         );
-      } catch (final KafkaException | KafkaResponseGetFailedException e) {
+        String serviceId = "default"; //FIXME not sure how to get this
+        if (sourceQueries.isEmpty()){
+          consumerGroupDescription = Optional.empty();
+        } else {
+          String queryId = sourceQueries.get(0).getId().toString();
+          String consumerGroupId = "_confluent-ksql-" + serviceId + "_" + KsqlConfig.KSQL_PERSISTENT_QUERY_NAME_PREFIX_DEFAULT + queryId; //FIXME there should be a better way to build this
+          consumerGroupDescription = Optional.of(
+              serviceContext.getAdminClient().describeConsumerGroups(Collections.singletonList(consumerGroupId)).describedGroups().get(consumerGroupId).get()
+          );
+          topicAndConsumerOffsets = serviceContext.getAdminClient().listConsumerGroupOffsets(consumerGroupId).partitionsToOffsetAndMetadata().get();
```

Comment:
as mentioned above, it would be nice to encapsulate this into the `KafkaTopicClient` (I think, I'm not 100% sure anymore - cc @big-andy-coates for his thoughts on this one)

Reply:
@agavra I went by adding methods to KafkaTopicClient, and elevate KafkaConsumerGroupClient&mdash;that was hidden in utils&mdash; to handle consumer group offsets query and add it to ServiceContext. Hope this looks better now.